internal/metrics/client.go

```diff
@@ -69,6 +78,45 @@ func NewClient(url string, timeout time.Duration) (Client, error) {
 	}, nil
 }
 
+func (c *client) Measure(b Benchmarker, f queryFunc, name string,job string, confDescription string, defaultRange model.Duration) error {
+	measure, err := f(job, defaultRange)
+	if err != nil {
+		return fmt.Errorf("queryMetric %s failed job: %s err: %w", name, job, err)
+	}
+	b.RecordValue(fmt.Sprintf("%s - %s - %s",job, name, confDescription), measure)
+	return nil
+}
+
+func (c *client) ProcessCPU(job string, duration model.Duration) (float64, error) {
+	query := fmt.Sprintf(
+		`sum(rate(process_cpu_seconds_total{job="%s"}[%s]) * 1000)`, // in m (i.e. 1000 = 1 vCore)
+		job, duration,
+	)
+
+	return c.executeScalarQuery(query)
+}
+
+// NOTE: ProcessWorkingSetMEM requires cadvisor to be deployed and functional
+func (c *client) ProcessWorkingSetMEM(caJob string, duration model.Duration) (float64, error) {
+	query := fmt.Sprintf(
+		`sum(avg_over_time(container_memory_working_set_bytes{job="%s"}[%s]) / 1000000)`, // in Mi (i.e. in Megabytes)
+		caJob, duration,
+	)
+
+	return c.executeScalarQuery(query)
+}
+
+// NOTE: Using ProcessResidentMEM is not recommended (information not representative for golang apps)
+// It is recommended to deploy cadvisor and use ProcessWorkingSetMEM
+func (c *client) ProcessResidentMEM(job string, duration model.Duration) (float64, error) {
+	query := fmt.Sprintf(
+		`sum(avg_over_time(process_resident_memory_bytes{job="%s"}[%s]) / 1000000)`, // in Mi (i.e. in Megabytes)
+		job, duration,
+	)
+
+	return c.executeScalarQuery(query)
+}
```

Comment:
Does this imply a fallback metric for target systems under test which don't provide cadvisor metrics? If not, should this be removed?

Reply:
@periklis Yes ... in the future I think we should add this as a fallback (also the same thing for memory)